src/flint/flint_base/flint_base.pyx

```diff
@@ -114,16 +119,117 @@ cdef class flint_poly(flint_elem):
                 v = - fac[0]
                 roots.append((v, m))
         return roots
-    
+
     def complex_roots(self):
         raise AttributeError("Complex roots are not supported for this polynomial")
 
 
+cdef class flint_mpoly_context(flint_elem):
+    """
+    Base class for multivariate ring contexts
+    """
+
+    _ctx_cache = None
+
+    def __cinit__(self):
+        self._init = False
+
+    def __init__(self, long nvars, names):
+        assert nvars >= 1
```

Comment:
I might be wrong but I believe that Flint's mpoly types support having zero variables. It would be good to support that here as well.

Reply:
I'll check that and get back to you. I also plan on removing all the asserts in favour of nicer error messages

Reply:
Yep looks like it's supported, https://github.com/flintlib/flint/blob/6fef2c48b/src/mpoly/ctx.c#L42-L44
I'll adjust this